src/main/java/pedido/Pedido.java

```diff
@@ -1,51 +1,113 @@
 package pedido;
 
+import ingredientes.Adicional;
+import ingredientes.Ingrediente;
+
 import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Map;
 
-public class Pedido{
+public class Pedido {
 
-    private int id;
-    private  ArrayList<ItemPedido> itens;
-    private Cliente cliente;
+    private final int id;
+    private final ArrayList<ItemPedido> itens;
+    private final Cliente cliente;
 
-    public Pedido(int id, ArrayList<ItemPedido> itens,Cliente cliente){
+    public Pedido(int id, ArrayList<ItemPedido> itens, Cliente cliente) {
         this.id = id;
-        this.itens=itens;
-        this.cliente=cliente;
+        this.itens = itens;
+        this.cliente = cliente;
     }
 
     public ArrayList<ItemPedido> getItens() {
         return itens;
     }
 
-    public int getId(){
+    public int getId() {
         return this.id;
     }
 
-    public Cliente getCliente(){
+    public Cliente getCliente() {
         return this.cliente;
     }
 
-    public double calcularTotal(Cardapio cardapio){
-        double total= 0;
-        //TODO
+    public double calcularTotal(Cardapio cardapio) {
+        Double total = (double) 0;
+        for (ItemPedido itemPedido : this.itens) {
+            Double totalItem = (double) 0;
+            for (Map.Entry<Ingrediente, Double> precoIngrediente : cardapio.getPrecos().entrySet()) {
+                if (itemPedido.getShake().getBase().compareTo(precoIngrediente.getKey()) == 0) {
+                    totalItem += precoIngrediente.getValue();
+                }
+            }
+            totalItem = totalItem * itemPedido.getShake().getTipoTamanho().multiplicador;
+            for (Map.Entry<Ingrediente, Double> precoIngrediente : cardapio.getPrecos().entrySet()) {
+                for (Adicional adicional : itemPedido.getShake().getAdicionais()) {
+                    if (adicional.equals(precoIngrediente.getKey())) {
+                        totalItem += precoIngrediente.getValue();
+                    }
+                }
+            }
+            totalItem = totalItem * itemPedido.getQuantidade();
+            total += totalItem;
+        }
```

Comment:
Aconselho você reduzir o uso de laços for, da para fazer essa lógica com stream, utilizando map e reduce. Vou anexar aqui para você um exemplo:
```
public double calcularTotal(final Cardapio cardapio) {
        return itens.stream()
                .map(itemPedido -> {
                    final var shake = itemPedido.getShake();
                    final var adiocinais = shake.getAdicionais();
                    final var precoBase = cardapio.buscarPreco(shake.getBase());
                    final var precoAdicionais = adiocinais.stream().map(cardapio::buscarPreco).reduce(0.0, Double::sum);
                    final var acrescimoBase = shake.getTipoTamanho().multiplicador * precoBase;
                    return (precoBase + acrescimoBase + precoAdicionais) * itemPedido.getQuantidade();
                })
                .reduce(0.0, Double::sum);
    }
```